### PR TITLE
:sparkles: Add topology field for MinReadySeconds

### PR DESCRIPTION
**Pull request by farodin91**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

**Comment by killianmuldoon** [comment]
Can you go into a bit of detail on what the goal of this change is? I don't think there's a way to set minReadySecond in the control plane as it's used in the MachineSet controller Control plane machines are created without a machineSet.

**Comment by farodin91** [comment]
@killianmuldoon I will remove it for control plane. I thought i saw it.

**Comment by farodin91** [comment]
@killianmuldoon Controlplane changes are removed

File: test/e2e/data/infrastructure-docker/v1beta1/main/bases/cluster-with-topology.yaml [hunk 1]
```diff
@@ -26,6 +26,7 @@ spec:
         name: "md-0"
         nodeDeletionTimeout: "30s"
         nodeVolumeDetachTimeout: "5m"
+        minReadySeconds: 30
```
Comment (sbueringer) [inline test/e2e/data/infrastructure-docker/v1beta1/main/bases/cluster-with-topology.yaml:29]:
```suggestion
        minReadySeconds: 5
```
nit. Not sure if that property infuences test speed in some way. Thus I would prefer a relatively low value

**Comment by sbueringer** [comment]
Thx for this PR. Looks good over. 

Looks like you have to run the following target to re-generate the OpenAPI spec
```
make generate-go-openapi
```

**Comment by farodin91** [comment]
@sbueringer I executed generate go openai, but it didn't work so i copied the patch from the build server.

**Comment by sbueringer** [comment]
Thx!

/lgtm
(assuming golangci-lint will be green in 2-3 min)

**Pull request merged**